### PR TITLE
DUOS-1072[risk=no]Dataset Update Validation Bug

### PR DESCRIPTION
**Pull request by raejohanek**
Scope:
This fixes the "complete all required fields" bug for all datasets, and the "Some errors occurred, the Dataset was not updated." bug that was appearing for a few datasets (447, 454)

Addresses:
https://broadworkbench.atlassian.net/browse/DUOS-1072

Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] PR is labeled with a Jira ticket number and includes a link to the ticket
- [ ] PR is labeled with a security risk modifier [no, low, medium, high] 
- [ ] PR describes scope of changes

In all cases:

- [ ] Get a minimum of one thumbs worth of review, preferably 2 if enough team members are available
- [ ] Get PO sign-off for all non-trivial UI or workflow changes
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by rushtong — CHANGES_REQUESTED** [review]
I'm still seeing a failure to update, but the error message looks incorrect as the POST operation succeeded. This still needs some debugging.

Screen Shot 2021-03-03 at 8 15 44 AM

File: src/pages/DatasetRegistration.js [hunk 1]
```diff
@@ -62,6 +62,7 @@ class DatasetRegistration extends Component {
         dac: '',
         consentId: '',
         needsApproval: false,
+        validName: false
```
Comment (rushtong) [inline src/pages/DatasetRegistration.js]: nit: rename to `isValidName`

**Comment by rushtong** [comment]
The `onNihStatusUpdate` and `nihValid` state variable look unused to me, we should clean that up in this PR to minimize whatever validation bugs there are.

**Comment by rushtong** [comment]
I think there is an API bug here as well. If we post a dataset with a valid dac id, it doesn't look like that is being honored. When retrieving the dataset again, the dac id is null.

**Comment by rushtong** [comment]
There is at least one API bug. We need to update this code: https://github.com/DataBiosphere/consent/blob/develop/src/main/java/org/broadinstitute/consent/http/service/DatasetService.java#L148

by adding this block to update the DAC on the consent
```
        if (Objects.nonNull(dataset.getDacId())) {
            Consent consent = consentDAO.findConsentFromDatasetID(datasetId);
            if (Objects.nonNull(consent)) {
                consentDAO.updateConsentDac(consent.getConsentId(), dataset.getDacId());
            }
        }
```

**Comment by rushtong** [comment]
Have confirmed locally that with the above fix ☝🏽  I can successfully update datasets and have them return with their correct DAC. This also fixes the initial load of the dataset in the case that it does have a DAC. Will file a separate PR for that.

**Comment by rushtong** [comment]
See https://github.com/DataBiosphere/consent/pull/956 for the API fix

**Comment by raejohanek** [comment]
> The `onNihStatusUpdate` and `nihValid` state variable look unused to me, we should clean that up in this PR to minimize whatever validation bugs there are.

the onNihStatusUpdate method is used by eraCommons

**Comment by rushtong** [comment]
@raejohanek 
> the onNihStatusUpdate method is used by eraCommons

`eraCommons` is not on the dataset registration page - I think this might have been a copy/paste issue.

**Review by rushtong — APPROVED** [review]
👍🏽

**Pull request merged**